projects/ui/src/lib/components/po-field/po-rich-text/po-rich-text-modal/po-rich-text-modal.component.ts

```diff
@@ -0,0 +1,128 @@
+
+const uploadRestrictions = ['.apng', '.bmp', '.gif', '.ico', '.jpeg', '.jpg', '.png', '.svg'];
+
+@Component({
+  selector: 'po-rich-text-modal',
+  templateUrl: './po-rich-text-modal.component.html'
+})
+export class PoRichTextModalComponent {
+
+  modalType: PoRichTextModalType;
+  savedCursorPosition;
+  selection = document.getSelection();
+  uploadDisabled = false;
+  uploadModel: Array<any>;
+  uploadRestrictions: PoUploadFileRestrictions = {
+    allowedExtensions: uploadRestrictions
+  };
+  urlImage: string;
+
+  readonly literals = {
+    ...poRichTextLiteralsDefault[this.languageService.getShortLanguage()]
+  };
+
+  modalCancelAction: PoModalAction = {
+    label: this.literals.cancel,
+    action: () => this.closeModal()
+  };
+
+  modalConfirmAction: PoModalAction = {
+    label: this.literals.insert,
+    disabled: false,
+    action: () => this.insertElementRef(this.urlImage)
+  };
+
+  @ViewChild('modal', { static: true }) modal: PoModalComponent;
+
+  @ViewChild('modalImageForm', { static: true }) modalImageForm: NgForm;
+
+  @ViewChild('upload', { static: true }) upload: PoUploadComponent;
+
+  @ViewChild('modalImage', { static: true }) modalImage: ElementRef;
+
+  @Output('p-command') command = new EventEmitter<string | { command: string, value: string }>();
+
+  constructor(private languageService: PoLanguageService) {}
+
+  async convertToBase64(urlImage?: string) {
+    const haveUploadFile = this.uploadModel && this.uploadModel.length;
+    if (!urlImage && haveUploadFile) {
+      const uploadImage = this.uploadModel[0].rawFile;
+      urlImage = await convertImageToBase64(uploadImage);
+      return urlImage;
+    }
+  }
+
+  emitCommand(value) {
+    let command: string;
+    if (value && this.modalType === PoRichTextModalType.Image) {
+      command = 'insertImage';
+      this.command.emit(({ command, value }));
+    }
+  }
+
+  async insertElementRef(urlImage?: string) {
+    if (this.modalType === PoRichTextModalType.Image && !urlImage) {
+      urlImage = await this.convertToBase64(urlImage);
+    }
+    this.closeModal();
+    this.retrieveCursorPosition();
+    this.emitCommand(urlImage);
+  }
+
+  openModal(type: PoRichTextModalType) {
+    this.modalType = type;
+    if (this.modalType === PoRichTextModalType.Image) {
+      this.uploadDisabled = false;
+    }
+    this.saveCursorPosition();
+    this.modal.open();
+  }
+
+  validateUrlImagelOnChange(form: NgForm) {
+    this.changeDisableUpload(form);
+  }
+
+  private changeDisableUpload(form: NgForm) {
+    return this.uploadDisabled = this.isUrlValid(form);
+  }
+
+  private cleanUpFields() {
+    this.urlImage = undefined;
+    this.uploadModel = undefined;
+    if (this.upload) {
+      this.uploadDisabled = false;
+    }
+  }
+
+  private closeModal() {
+    this.cleanUpFields();
+    this.modal.close();
+  }
+
+  private isUrlValid(form: NgForm): boolean {
+    if (form) {
+      return this.urlImage && this.urlImage !== '' && !form.invalid;

```

Comment:
O `this.urlImage !== ''` poderia ser removido porque a condição `this.urlImage` já valida se a string não é vazia também :)